File: Subforo.js

function Subforo(nombre, id, descripcion, creador, fecha,
                moderadores, temas) {
    let nombre = nombre;
    let id = id;
    let descripcion = descripcion;
    let creador = creador;
    let fecha = fecha;
    let moderadores = moderadores;
    let temas = temas;

    this.getNombre = function() {
        return this.nombre;
    }
    this.getDescripcion = function() {
        return this.descripcion;
    }
    this.getId = function() {
        return this.id;
    }
    this.getCreador = function() {
        return this.creador;
    }
    this.getFecha = function() {
        return this.fecha;
    }
}

File: src/main/java/com/metodologia/foro/entities/Subforo.java
package com.metodologia.foro.entities;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "subforos")
public class Subforo {
    private int id;
	private String titulo;
	private String descripcion;
	private Date fecha;
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", unique = true, nullable = false)
    public int getId() {
        return this.id;
    }

    @Column(name = "titulo", nullable = false, length = 30)
    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    @Column(name = "descripcion", nullable = false, length = 30)
    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    @Column(name = "fecha", nullable = false)
    public Date getFechaSubscripcion() {
        return fecha;
    }
}
